### ISSUE TITLE
Data null in background

### ISSUE THREAD

**Issue by gibrandev** [zo0r/react-native-push-notification#1627]
# Bug
Data payload null if in background
In Background
```json
{"chat": "us0jXE4qG9gPuSMJF0xzHTH", "collapse_key": "com.hallopet.user", "data": {}, "finish": [Function finish], "foreground": false, "from": "1095807157509", "google.delivered_priority": "normal", "google.message_id": "0:1598946632974825%eaee66e3eaee66e3", "google.original_priority": "normal", "google.sent_time": 1598946632971, "google.ttl": 2419200, "object": "chat", "object_id": "us0jXE4qG9gPuSMJF0xzHTH", "userInteraction": true}
```
In Active App
```json
{"color": null, "data": {"chat": "us0jXE4qG9gPuSMJF0xzHTH", "object": "chat", "object_id": "us0jXE4qG9gPuSMJF0xzHTH"}, "finish": [Function finish], "foreground": true, "id": "1860630111", "message": "Pp", "sound": null, "title": "cepw4xh5n8", "userInteraction": false}
```

## Environment info

`react-native info` output:

```bash
System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 1.31 GB / 7.91 GB
  Binaries:
    Node: 14.9.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.4 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.8 - C:\Program Files\nodejs\npm.CMD
  IDEs:
    Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6626763
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: 0.61.5 => 0.61.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

Library version: ^5.1.0

Describe what you expected to happen:

1. I will data is not null like Active App notification attached

## Reproducible sample code

```javascript
import React, { Component } from 'react'
import Route from './src/Route';
import { Root } from 'native-base';
import AsyncStorage from '@react-native-community/async-storage'
import axios from 'axios'
import SplashScreen from 'react-native-splash-screen'
import { AppState, Text, View, Image, StatusBar, Dimensions, StyleSheet, Linking, Platform } from 'react-native'
import { API_URL, BASE_COLOR, SOCKET_URL } from 'react-native-dotenv'
import Icon from 'react-native-vector-icons/AntDesign';
import NotifService from './NotifService'; // use example of this library
import NavigationService from './NavigationService';  // use example of this library
import ADD_DATA from './redux/notification/action';
import storeNotification from './redux/notification/store';
import * as storeUser from 'react-native-simple-store';
import PushNotificationIOS from "@react-native-community/push-notification-ios";

import { Provider} from 'react-redux';
import store from './redux/store';

export class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showRealApp: true
        };
        this.notif = new NotifService(
            this.onRegister.bind(this),
            this.onNotification.bind(this),
        );
    }
    onRegister(token) {
        AsyncStorage.setItem('notification:token', token.token);
    }
    onNotification(notif){
        console.log(notif);
    }
    componentDidMount() {
        SplashScreen.hide();
    }
    render() {
            return (
                <Root>
                    <Provider store={store}>
                        <Route />
                    </Provider>
                </Root>
            );
    }
}

export default App
```

**Comment by Dallas62** [comment]
Hi,

Without a reproducible example, we are not able to look into this issue.

Regards

**Comment by gibrandev** [comment]
i have updated my issue

**Comment by Dallas62** [comment]
Can you provide what is send by the server ?

**Comment by gibrandev** [comment]
I send in the server like this. Data and notification only with package: https://github.com/kreait/laravel-firebase
```php
$notification = Notification::fromArray([
    'title' => 'System',
    'body' => 'Welcome'
]);
$data = [
    'id' => 1,
    'object' => 'welcome',
    'object_id' => 1,
];
$message = CloudMessage::new()
    ->withNotification($notification)
    ->withData($data);
```

**Comment by gibrandev** [comment]
FYI in ios work perfect.

**Comment by Dallas62** [comment]
Hi @gibrandev 
Will try to provide you a fix on `dev` branch in the next 2 hours.
Regards

**Comment by gibrandev** [comment]
ok i will waiting for this. thanks

**Comment by Dallas62** [comment]
I don't have an iOS application setup for test, can you provide the payload returned by `onNotification` on iOS ?
By doing the exact same test (background + user click).

**Comment by gibrandev** [comment]
In Active
```json
{"badge": undefined, "data": {"chat": "vYxVH1ADEhuSLFOsVwo6niGUnqiiqyKN", "gcm.message_id": "1599034437820075", "google.c.a.e": "1", "notificationId": "691D52E8-DFFE-4B39-A31F-1B711073AA17", "object": "chat", "object_id": "vYxVH1ADEhuSLFOsVwo6niGUnqiiqyKN", "remote": true, "userInteraction": 0}, "finish": [Function finish], "foreground": true, "id": undefined, "message": {"body": "Pp", "title": "gibrandev"}, "soundName": undefined, "title": undefined, "userInteraction": false}
```
In Background
```json
{"badge": undefined, "data": {"chat": "vYxVH1ADEhuSLFOsVwo6niGUnqiiqyKN", "gcm.message_id": "1599034488426493", "google.c.a.e": "1", "notificationId": "37BA650A-5458-480F-94C4-9452608C2272", "object": "chat", "object_id": "vYxVH1ADEhuSLFOsVwo6niGUnqiiqyKN", "remote": true, "userInteraction": 1}, "finish": [Function finish], "foreground": false, "id": undefined, "message": {"body": "Pp", "title": "gibrandev"}, "soundName": undefined, "title": undefined, "userInteraction": true}
```

**Comment by Dallas62** [comment]
The fix is available on `dev` branch.

**Comment by gibrandev** [comment]
now work with branch dev. Thanks @Dallas62

**Comment by Dallas62** [comment]
Good news, will release the new version soon. Will close the issue after 😉

**Comment by Dallas62** [comment]
You can test changes on `dev` branch.

Please refer to the CHANGELOG for update instructions:
https://github.com/zo0r/react-native-push-notification/blob/dev/CHANGELOG.md#unreleased

To install this version:
```
npm install zo0r/react-native-push-notification#dev
```
OR
```
yarn add zo0r/react-native-push-notification#dev
```

Make sure your build cache is cleaned.

**Comment by gibrandev** [comment]
@Dallas62 fixed in version 6.0.0, can i close this issue
